FlowCrypt/Controllers/Attachment/AttachmentViewController.swift

```
@@ -59,21 +60,29 @@ final class AttachmentViewController: UIViewController {
         saveAndStartDownload(file: file)
     }
 
-    override func viewWillDisappear(_ animated: Bool) {
-        super.viewWillDisappear(animated)
-        remove(file: file)
-    }
-
     private func remove(file: FileType) {
-        Task {
-            try await filesManager.remove(file: file)
+        Task { [weak self] in
+            do {
+                try await filesManager.remove(file: file)
+                self?.navigationController?.popViewController(animated: true)
+            } catch {
+                showAlert(message: "could_not_remove_opened_attachment".localized) {
+                    self?.navigationController?.popViewController(animated: true)
+                }
+            }
         }
     }
 
     private func saveAndStartDownload(file: FileType) {
         Task { [weak self] in
-            let url = try await filesManager.save(file: file)
-            self?.load(with: url)
+            do {
+                let url = try await filesManager.save(file: file)
+                self?.load(with: url)
+            } catch {
+                showAlert(message: "could_not_open_an_attachment".localized) {
```

Same as above, don't swallow error details: surface them for easier debugging.